lighthouse-core/gather/gather-runner.js

```diff
@@ -197,6 +200,11 @@ class GatherRunner {
                 // Merge pass trace and network data into tracingData.
                 config.trace && Object.assign(tracingData.traces, loadData.traces);
                 config.network && (tracingData.networkRecords = loadData.networkRecords);
+              })
+              .then(_ => {
+                // Update the outer options object of any post-redirect url
+                // Subsequent passes will use the new URL, and it'll be reported outwards
+                options.url = runOptions.url;
```

Comment:
seems like this could bite us, though. Someone adding a new first pass, moving the performance gatherers to the second pass, would miss the redirects in the load time since the second pass would skip to the new URL. Is there a reason each pass shouldn't start with the same URL each time?